tls/s2n_server_key_exchange.c

```diff
@@ -135,6 +136,42 @@ int s2n_dhe_server_key_recv_parse_data(struct s2n_connection *conn, union s2n_ke
     return 0;
 }
 
+int s2n_kem_server_key_recv_read_data(struct s2n_connection *conn, struct s2n_blob *data_to_verify, union s2n_kex_raw_server_data *raw_server_data)
+{
+    struct s2n_kem_raw_server_params *kem_data = &raw_server_data->kem_data;
+    struct s2n_stuffer *in = &conn->handshake.io;
+    const struct s2n_kem *kem = conn->secure.kem_params.negotiated_kem;
+    uint16_t key_length;
+
+    /* Keep a copy to the start of the whole structure for the signature check */
+    data_to_verify->data = s2n_stuffer_raw_read(in, 0);
+    notnull_check(data_to_verify->data);
+
+    // the server sends the named KEM again and this must match what was agreed upon during server hello
+    uint8_t named_kem;
+    GUARD(s2n_stuffer_read_uint8(in, &named_kem));
+    eq_check(named_kem, kem->kem_extension_id);
+
+    GUARD(s2n_stuffer_read_uint16(in, &key_length));
+    S2N_ERROR_IF(key_length > s2n_stuffer_data_available(in), S2N_ERR_BAD_MESSAGE);
+
+    kem_data->raw_public_key.data = s2n_stuffer_raw_read(in, key_length);
+    notnull_check(kem_data->raw_public_key.data);
+    kem_data->raw_public_key.size = key_length;
+
+    data_to_verify->size = KEM_EXTENSION_BYTES + KEM_PUBLIC_KEY_BYTES + key_length;
+
+    return 0;
+}
+
+int s2n_kem_server_key_recv_parse_data(struct s2n_connection *conn, union s2n_kex_raw_server_data *raw_server_data)
+{
+    struct s2n_kem_raw_server_params *kem_data = &raw_server_data->kem_data;
+    conn->secure.kem_params.public_key.data = kem_data->raw_public_key.data;
```

Comment:
Is there an s2n_blob_copy function? what will happen if an s2n_blob has/will have additional parameters to data and size? 
Does this copy data&size pattern happens again in s2n? If so, maybe consider add an s2n_blob_copy, which will reduce LOCs.

What about checking for null blobs?

Reply:
Decided to go with `s2n_dup` (pending further research and benchmarking) for now since this is a drop compared to the rest of the BIKE/SIKE calculation.